### PR TITLE
Add ENV level disable via DISABLE_MINITEST_SILENCE

### LINKED ISSUE #3: Does not work with interactive debuggers
Adding `byebug` or `binding.irb` to a test does not work with this plugin - the test suite appears to hang, and is not interruptible. Using `--verbose` doesn't help because that output is buffered.

Ideally we could automatically detect this kind of usage and allow it to work, but if not it might be nice to provide a way to temporarily disable the plug-in, or at least allow interrupts to stop the runner.

### PR DESCRIPTION
**Pull request by gmcgibbon**
Closes #3 

Adds support for an ENV level disable. I'm open to suggestions for alternatives, but adding what I outline in the readme should fix the common use-case for local development.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,8 @@ The plugin will be automatically activated by Minitest if it is in your applicat
 
 You can disable the plugin by providing the `--disable-silence` command line option to your test invocation. The primary use case for this is when you want to use a debugger inside a test, which will require a the standard input and outputs to work interactively.
 
+Alternattively, you can use `ENV["DISABLE_MINITEST_SILENCE"]` to selectively disable the plugin. For example, if a Rails application always wants to disable silencing in local test runs, `ENV["DISABLE_MINITEST_SILENCE"] = "yes" unless ENV["CI"]` will do that.
```
Comment (JaKXz) [inline README.md:29]:
```suggestion
Alternatively, you can use `ENV["DISABLE_MINITEST_SILENCE"]` to selectively disable the plugin. For example, if a Rails application always wants to disable silencing in local test runs, `ENV["DISABLE_MINITEST_SILENCE"] = "yes" unless ENV["CI"]` will do that.
```

**Review by JaKXz — COMMENTED** [review]
This makes sense, but I think that we want `ENV["DISABLE_MINITEST_SILENCE"] = "yes" unless ENV["CI"]` set internally by default so I think it would be worth making disabling minitest silence unless in CI the default behaviour.

**Comment by gmcgibbon** [comment]
I changed it to disable if `ENV["CI"]` is unset, but this will cause trouble for people who want this locally. If that use-case exists.

**Comment by gmcgibbon** [comment]
Alright, it is opt-in now. Any of these solutions work, but this is a breaking change. Feedback would be appreciated.

**Review by rafaelfranca — APPROVED** [review]

**Comment by rafaelfranca** [comment]
This is pre 1.0, so breaking changes are fine. I think it would be only breaking to people that were using --disable_silence

**Pull request merged**